src/exception.cr

```diff
@@ -129,6 +129,18 @@ class DivisionByZeroError < Exception
   end
 end
 
+# Raised when the result of an arithmetic operation is outside of the range
+# that can be represented within the given operands types.
+#
+# ```
+# Int32::Max + 1 # raises OverflowError (Overflow)
+# ```
+class OverflowError < Exception
+  def initialize(message = "Overflow")
```

Comment:
Maybe this should be "Overflow or underflow"? (unless we have a way to detect whether it was overflow or underflow, but I don't think so)

Reply:
And maybe also be a bit more specific about what kind of overflow it is?
For example `Integer overflow or underflow`.

Reply:
Underflow occur on floating point numbers. I'm ok to use the same exception. But we are not checking for underflow here.

If we aim to use the same exception, the proposal is to change the message or the name also?

Reply:
I mean, when you subtract one from the most negative integer, is that an underflow or an overflow?

Reply:
Nevermind, it seems that's an overflow too. Ignore this comment :-)